exonum/src/blockchain/block.rs

```diff
@@ -62,6 +102,63 @@ pub struct Block {
     /// These calls can include transactions, `before_transactions` and/or `after_transactions` hooks
     /// for services.
     pub error_hash: Hash,
+    /// Additional information that can be added into the block.
+    pub entries: BlockHeaderEntries,
+}
+
+impl Block {
+    /// Insert new entry to the block header.
+    pub(crate) fn add_entry<K: BlockHeaderKey>(&mut self, value: K::Value) {
+        self.entries.insert::<K>(value);
+    }
+
+    /// Get block header entry for specified key type. Key type is specified via
+    /// type parameter.
+    ///
+    /// # Usage
+    ///
+    /// ```
+    /// use exonum::crypto::Hash;
```

Comment:
It's [possible](https://doc.rust-lang.org/rustdoc/documentation-tests.html#hiding-portions-of-the-example) to hide portions of the test by starting a line with `# ` (which is already used in some doc tests in Exonum). Here, it could make sense to hide imports, `impl BinaryValue`, `Block` fields other than `entries`, etc. It could also help commenting before the example or in the example what `ActiveServices` is; something like: "Suppose we store a list of active service IDs in a block. We can do this by defining a corresponding `BlockHeaderKey` implementation" and so on.